### DISCUSSION TITLE
How we can use separate endpoints for creating and updating objects in FastAPI

### DISCUSSION THREAD

**Discussion by mojixcoder** [art049/odmantic#253] [Q&A]
I want to be able to update and create instances with two endpoints.  

does `odmantic` has any built-in ways to do it?  

I mean i want to not to let the client to give me the id of object for creating instances.

and for updating instances client must give me an id.  

is it possible via `odmantic` models? or i should write new `pydantic` models to do this?

**Comment by hellocoldworld** [comment]
Are you using FastAPI? The docs have an [example](https://art049.github.io/odmantic/usage_fastapi/#example) of how to work with FastAPI and have an example for creating instances and one for updating instances.

**Comment by mojixcoder** [comment]
The AIOEngine.save and AIOEngine.save_all methods are actually behaving as upsert operations. In other words, if the instance already exists it will be updated. Otherwise, the related document will be created in the database.
What I want is to turn off this behaviour. I want to raise an exception for updating instances that do not exist. I know I can do it manually but I was looking for a built-in solution. @hellocoldworld
